packages/components/src/components/chip/chip-item/chip-item.tsx

```diff
@@ -0,0 +1,158 @@
+
+    /**
+     * Listenting for ifxChipItem event to unselect the previously selected Item.
+     * Only used for the *single* variant of Chip.
+     */
+    @Listen('ifxChipItem', { target: 'body' })
+    updateItemSelection(event: CustomEvent<ChipItemEvent>) {
+        if (this.chipState.variant === 'single') {
+            const target = event.target as HTMLIfxChipItemElement;
+            /* Also making sure chip items are from the same group (parent) while unselecting. */
+            if (this.chipItem !== target && this.chipItem.parentElement === target.parentElement) {
+                this.selected = false;
+            }
+        }
+    } 
+
+    /**
+     * Keeping internal *selected* state in sync with the selected prop if changed.
+     */
+    @Watch('selected')
+    validateSelected(newValue: boolean, oldValue: boolean) {
+        if (newValue !== oldValue) {
+            /* Do not emit if ChipState does not allow. */
+            if (this.chipState.emitIfxChipItem){
+                this.emitIfxChipItemEvent();
+            } else {
+                this.chipState.emitIfxChipItem = true;
+            }
+        }
+    } 
+    
+    /**
+     * A function extracts and returns the Item label.
+     * 
+     * @returns string
+     */
+    getItemLabel(): string {
+        return this.chipItem.innerText as string;
+    }
+
+    /**
+     * Toggles the selection of the Chip Item and emits the ifxChipItem event.
+     */
+    toggleItemSelection() {
+        this.selected = !this.selected;
+    }
+        
+    /**
+     * Helper functions
+     */
+
+    emitIfxChipItemEvent(emitIfxChange: boolean = true) {
+        this.ifxChipItem.emit({ emitIfxChange: emitIfxChange,
+                                key: this.chipState.key,
+                                label: this.getItemLabel(), 
+                                selected: this.selected, 
+                                value: this.value });
+    }
+
+    handleItemClick() {
+        this.toggleItemSelection();
+    }
+
+    handleItemKeyDown(event: KeyboardEvent) {
+        if (event.code === 'Enter' || event.code === 'Space') {
+            this.toggleItemSelection();
+        }
+    }
+
+    handleSelectedState() {
+        if (this.selected) {
+            this.emitIfxChipItemEvent(false);
+        }
+    }
+
+    /**
+     * Lifecycle Methods
+     */
+
+    componentWillLoad() {
+        /* Propogating the selected state to the Chip (Parent) component if it is set. */
+        this.handleSelectedState();
+    }
+
+    render() {
+        return (
+            <div class={`chip-item chip-item--${this.chipState.size} 
+                         chip-item--${(this.selected && this.chipState.variant) === 'single' ? 'selected' : ''}`} 
+                tabIndex={0}
+                onClick={() => {this.handleItemClick()}}
+                onKeyDown={(e) => {this.handleItemKeyDown(e)}}>
+
+                {/* Checkbox; renders only in 'multi' variant. */}
```

Comment:
can you explain the reason for why you use checkbox + label here and what was the advantage over using multiselect for example?